src/coreclr/vm/dynamicmethod.h

```diff
@@ -331,25 +331,40 @@ inline MethodDesc* GetMethod(CORINFO_METHOD_HANDLE methodHandle)
 
 #ifndef DACCESS_COMPILE
 
-#define CORINFO_MODULE_HANDLE_TYPE_MASK 1
-
 enum CORINFO_MODULE_HANDLE_TYPES
 {
-    CORINFO_NORMAL_MODULE  = 0,
-    CORINFO_DYNAMIC_MODULE,
+    // The module handle is a Module
+    CORINFO_NORMAL_MODULE    = 0,
+
+    // The module handle is a DynamicResolver
+    CORINFO_DYNAMIC_MODULE   = 1,
+
+    // The module handle permits unrestricted access
+    CORINFO_MODULE_ALLACCESS = 2,
 };
 
+#define CORINFO_MODULE_HANDLE_TYPE_MASK (CORINFO_NORMAL_MODULE | CORINFO_DYNAMIC_MODULE | CORINFO_MODULE_ALLACCESS)
+
 inline bool IsDynamicScope(CORINFO_MODULE_HANDLE module)
 {
     LIMITED_METHOD_CONTRACT;
-    return (CORINFO_DYNAMIC_MODULE == (((size_t)module) & CORINFO_MODULE_HANDLE_TYPE_MASK));
+    return !!(CORINFO_DYNAMIC_MODULE & (((size_t)module) & CORINFO_MODULE_HANDLE_TYPE_MASK));
+}
+
+inline bool IsAllAccessScope(CORINFO_MODULE_HANDLE module)
+{
+    LIMITED_METHOD_CONTRACT;
+    return !!(CORINFO_MODULE_ALLACCESS & (((size_t)module) & CORINFO_MODULE_HANDLE_TYPE_MASK));
```

Comment:
```suggestion
    return IsDynamicScope(module) && GetDynamicResolver(module)->SuppressVisibilityChecks();
```
This would be cleaned than the bit stealing.